diff --git a/examples/instancing/instancing.cpp b/examples/instancing/instancing.cpp
index 0cdc9f0..fdc8295 100644
--- a/examples/instancing/instancing.cpp
+++ b/examples/instancing/instancing.cpp
@@ -9,10 +9,21 @@
 #include "vulkanexamplebase.h"
 #include "VulkanglTFModel.h"
 
+#include "C:/Users/Constantine/Desktop/vkfast/vkfast_ex.h"
+#include "C:/Users/Constantine/Desktop/vkfast/vkfast_ids.h"
+#include "C:/Users/Constantine/Desktop/vkfast/extra/Banzai/vkfast_extra_banzai_pointer.h"
+#include "C:/Users/Constantine/Desktop/vkfast/extra/REII/vkfast_extra_reii.h"
+#define VKFAST_EXAMPLES_COMMON_DO_NOT_LINK_GLFW3
+#define VKFAST_EXAMPLES_COMMON_INCLUDE_GLFW3
+#define VKFAST_EXAMPLES_COMMON_INCLUDE_EXTRA_BANZAI
+#include "C:/Users/Constantine/Desktop/vkfast/examples/Common/vkfast_examples_common.h"
+#pragma comment(lib, "C:/Users/Constantine/Desktop/vkfast/extra/Build vkFast as a library/vkFast.lib")
+#include <glm/gtx/matrix_decompose.hpp>
+
 #if defined(__ANDROID__)
 #define INSTANCE_COUNT 4096
 #else
-#define INSTANCE_COUNT 8192
+#define INSTANCE_COUNT 700000
 #endif
 
 class VulkanExample : public VulkanExampleBase
@@ -70,10 +81,10 @@ public:
 	VulkanExample() : VulkanExampleBase()
 	{
 		title = "Instanced mesh rendering";
-		camera.type = Camera::CameraType::lookat;
+		camera.type = Camera::CameraType::firstperson;
 		camera.setPosition(glm::vec3(5.5f, -1.85f, -18.5f));
 		camera.setRotation(glm::vec3(-17.2f, -4.7f, 0.0f));
-		camera.setPerspective(60.0f, (float)width / (float)height, 1.0f, 256.0f);
+		camera.setPerspective(90.0f, (float)width / (float)height, 1.0f, 256.0f);
 	}
 
 	~VulkanExample()
